File: tests/fixtures/fullstack/server/routes/language/yon.js
// @ts-check
//
// GET/HEAD /language — the JavaScript face of the consolidated polyglot route.
// The same route path is also served by yon.rs (POST, DELETE) and yon.cpp
// (PUT, PATCH), demonstrating one Yon route split across three languages.

export class Handler {
  /** @returns {Record<string, unknown>} */
  static GET() {
    return {
      route: '/language',
      description: 'One route, three languages: JavaScript answers GET/HEAD, Rust answers POST/DELETE, C++ answers PUT/PATCH.',
      methods: {
        GET: 'JavaScript route map (this response)',
        HEAD: 'JavaScript headers-only probe',
        POST: 'Rust echo handler compiled in-house to WebAssembly',
        DELETE: 'Rust size-confirm handler compiled in-house to WebAssembly',
        PUT: 'C++ echo handler compiled in-house to WebAssembly',
        PATCH: 'C++ patch-summary handler compiled in-house to WebAssembly',
      },
      routes: [
        { path: '/language', purpose: 'polyglot method split across JavaScript, Rust and C++' },
        { path: '/language/javascript', purpose: 'status-code diagnostics and echo responses' },
        { path: '/language/python', purpose: 'Python GET handler' },
        { path: '/language/rust', purpose: 'Rust GET handler' },
        { path: '/language/items', purpose: 'RESTful CRUD in TypeScript' },
        { path: '/language/items/:id', purpose: 'single-item CRUD in TypeScript' },
        { path: '/language/versions/:version', purpose: 'dynamic path parameter in Python' },
        { path: '/language/fylo', purpose: 'FYLO machine-interface showcase' },
        { path: '/language/telemetry', purpose: 'OTLP trace summary from FYLO' },
      ],
    }
  }

  /** @returns {Record<string, never>} */
  static HEAD() {
    return {}
  }
}

File: tests/fixtures/fullstack/server/routes/language/yon.rs
// POST/DELETE /language — the Rust face of the consolidated polyglot route.
//
// One route, three languages: yon.js answers GET/HEAD and yon.cpp answers
// PUT/PATCH beside this file. Yon generates the dependency-free `YonJson`
// adapter, so the handler needs no third-party crates.

struct Handler;

impl Handler {
    // POST — echo the JSON request body back as a structured result.
    pub fn POST(request: &YonJson) -> YonJson {
        let body = match request.get("body") {
            Some(value) => value.clone(),
            None => YonJson::Null,
        };
        YonJson::object(vec![
            ("language", "rust".into()),
            ("action", "echo".into()),
            ("body", body),
        ])
    }

    // DELETE — confirm how many bytes the request body carried.
    pub fn DELETE(request: &YonJson) -> YonJson {
        let size = match request.get("body") {
            Some(value) => value.stringify().len(),
            None => 0,
        };
        YonJson::object(vec![
            ("language", "rust".into()),
            ("action", "size-confirm".into()),
            ("receivedBytes", size.into()),
        ])
    }
}
